service_impl/bucketTopologyService.go

```diff
@@ -443,6 +443,28 @@ func (b *BucketTopologyService) getLocalBucketTopologyUpdater(spec *metadata.Rep
 			// This shouldn't happen.
 			watcher.logger.Errorf("%v Failed to get source %v. Error=%v", spec.SourceBucketName, base.CollectionsManifestUidKey, err.Error())
 		}
+
+		crossClusterVer, err := b.utils.GetCrossClusterVersioningFromBucketInfo(bucketInfo)
+		if err != nil {
+			// This shouldn't happen.
+			watcher.logger.Errorf("%v Failed to get source %v. Error=%v", spec.SourceBucketName, base.EnableCrossClusterVersioningKey, err.Error())
```

Comment:
Are you sure this is going to be present in mixed mode?
If we are in the middle of upgrading, is there a chance the key does not exist when one legacy node is still in the cluster
Then, this will be spamming the logs

Reply:
Good point. I will remove the log spam.